src/core/dev_api/shape_util.hpp

```diff
@@ -0,0 +1,28 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/shape.hpp"
+
+namespace ov {
+namespace util {
+/**
+ * @brief Makes spacial version of 2D ov::Shape which is recognize as dynamic.
+ *
+ * This is special case used for tensor <-> host tensor conversion to indicate that tensor got dynamic shape.
+ *
+ * @return 2-D shape with {0, SIZE_MAX}
+ */
+OPENVINO_API Shape make_dynamic_shape();
```

Comment:
Can we immediately mark these methods as deprecated in order to remove together with legacy API?

Reply:
Yes, marked and now is core internal only. Looks is not required by other libraries.